apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/WebsiteMonitoringCaseActions/hooks/useWebsiteMonitoringCaseActions/useWebsiteMonitoringCaseActions.ts

```diff
@@ -0,0 +1,60 @@
+import { useAuthenticatedUserQuery } from '@/domains/auth/hooks/queries/useAuthenticatedUserQuery/useAuthenticatedUserQuery';
+import { useApproveCaseMutation } from '@/domains/entities/hooks/mutations/useApproveCaseMutation/useApproveCaseMutation';
+import { useRejectCaseMutation } from '@/domains/entities/hooks/mutations/useRejectCaseMutation/useRejectCaseMutation';
+import { useCaseState } from '@/pages/Entity/components/Case/hooks/useCaseState/useCaseState';
+import { useEntityLogic } from '@/pages/Entity/hooks/useEntityLogic/useEntityLogic';
+import { StateTag } from '@ballerine/common';
+import { useCallback, useState } from 'react';
+
+export const useWebsiteMonitoringCaseActionsLogic = () => {
+  const { workflow } = useEntityLogic();
+  const { data: session } = useAuthenticatedUserQuery();
+  const authenticatedUser = session?.user;
+  const caseState = useCaseState(authenticatedUser, workflow);
+
+  const { mutate: onMutateApproveEntity, isLoading: isApproveCaseLoading } = useApproveCaseMutation(
+    { workflowId: workflow?.id as string },
+  );
+  const { mutate: onMutateRejectEntity, isLoading: isRejectCaseLoading } = useRejectCaseMutation({
+    workflowId: workflow?.id as string,
+  });
+
+  const [formValues, setValues] = useState({
+    action: '',
+    comment: '',
+  });
+
+  const handleResolutionActionChange = useCallback(
+    (value: string) => setValues(prev => ({ ...prev, action: value })),
+    [],
+  );
+
+  const handleResolutionCommentChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setValues(prev => ({ ...prev, comment: event.target.value }));
+    },
+    [],
+  );
+
+  const handleSubmit = useCallback(() => {
```

Comment:
Maybe we should throw an error or log something if `formValues.action` is invalid?

Reply:
Added toast